pnpm/src/main.ts

```diff
@@ -200,7 +200,7 @@ export async function main (inputArgv: string[]): Promise<void> {
     const filterResults = await filterPackagesFromDir(wsDir, filters, {
       engineStrict: config.engineStrict,
       nodeVersion: config.nodeVersion ?? config.useNodeVersion,
-      patterns: cliOptions['workspace-packages'] ?? config.workspaceManifest?.contents.packages,
+      patterns: cliOptions['workspace-packages'] ?? config.workspaceManifest?.contents.packages ?? ['.', '**'],
```

Comment:
This is one of the cases where a missing `pnpm-workspace.yaml` actually means everything is part of the workspace.

Is it intended that `pnpm install --recursive` treats all folders with `package.json` files as workspace packages even when `pnpm-workspace.yaml` isn't on disk? That's the current behavior here. I added `?? ['.', '**']` to make this more clear and to satisfy the required `patterns` argument.